terraform/ec2/mac/main.tf

```diff
@@ -98,28 +98,38 @@ resource "null_resource" "integration_test" {
       "sudo softwareupdate --install-rosetta --agree-to-license",
       "sudo curl https://awscli.amazonaws.com/AWSCLIV2.pkg -o AWSCLIV2.pkg",
       "sudo installer -pkg AWSCLIV2.pkg -target /",
-      #Install Golang
-      "mkdir homebrew && curl -L https://github.com/Homebrew/brew/tarball/master | tar xz --strip 1 -C homebrew",
-      "homebrew/bin/brew install go",
-    ]
-  }
-  # Install agent binaries
-  provisioner "remote-exec" {
-    inline = [
+
+      #Install Brew and set path for mac1 and mac2 instances
+      "NONINTERACTIVE=1 /bin/bash -c \"$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)\"",
+      "(echo; echo 'eval \"$(/usr/local/bin/brew shellenv)\"') >> /Users/ec2-user/.zprofile",
+      "eval \"$(/usr/local/bin/brew shellenv)\"",
+      "(echo; echo 'eval \"$(/opt/homebrew/bin/brew shellenv)\"') >> /Users/ec2-user/.zprofile",
+      "eval \"$(/opt/homebrew/bin/brew shellenv)\"",
+
+      #Download test repo
+      "NONINTERACTIVE=1 brew install git",
+      "echo clone test repo",
+      "git clone --branch ${var.github_test_repo_branch} ${var.github_test_repo}",
+
+      # Install agent and validator binaries
+      "echo Install agent and validator binaries",
       "/usr/local/bin/aws s3 cp s3://${var.s3_bucket}/integration-test/packaging/${var.cwa_github_sha}/${var.arc}/amazon-cloudwatch-agent.pkg .",
       "/usr/local/bin/aws s3 cp s3://${var.s3_bucket}/integration-test/validator/${var.cwa_github_sha}/darwin/${var.arc}/validator .",
       "sudo installer -pkg amazon-cloudwatch-agent.pkg -target /",
-    ]
-  }
 
-  #Prepare the requirement before validation and validate the metrics/logs/traces
-  provisioner "remote-exec" {
-    inline = [
-      "sudo chmod +x ./validator",
+      # Install Golang
+      "echo Install golang",
+      "NONINTERACTIVE=1 brew install go",
```

Comment:
There's no guarantee that this is the right version of Go. What is in the brew cask right now?

Reply:
It's currently `go 1.20.5`. Do we need to specify the go version to 1.18?